core/src/main/scala/io/finch/route/Extractor.scala

```diff
@@ -13,37 +13,64 @@ case class Extractor[A](name: String, f: String => A) extends Router[A] { self =
       aa <- Try(f(ss)).toOption
     } yield (input.drop(1), () => Future.value(aa))
 
-  def apply(n: String): Extractor[A] = copy[A](name = n)
+  def apply(n: String): Router[A] = copy[A](name = n)
 
-  def * : Router[Seq[A]] = new Router[Seq[A]] {
-    override def apply(input: Input): Option[(Input, () => Future[Seq[A]])] =
-      Some((input.copy(path = Nil), () => Future.value(for {
-        s <- input.path
-        a <- Try(self.f(s)).toOption
-      } yield a)))
+  override def toString: String = s":$name"
+}
+
+/**
+ * An extractor that extracts a value of type `Seq[A]` from the tail of the route.
+ */
+case class TailExtractor[A](name: String, f: String => A) extends Router[Seq[A]] {
+  import Router._
+  def apply(input: Input): Option[(Input, () => Future[Seq[A]])] =
+    Some((input.copy(path = Nil), () => Future.value(for {
+      s <- input.path
+      a <- Try(f(s)).toOption
+    } yield a)))
 
-    override def toString = self.toString + ".*"
-  }
+  def apply(n: String): Router[Seq[A]] = copy[A](name = n)
 
-  override def toString: String = s":$name"
+  override def toString: String = s":$name*"
```

Comment:
Hm. I was thinking about that too, but decided to go with `*` since a) we already have matcher `*` that somehow matches the tail b) visually it's easy to misread `ints` as `int`, while start (`*`) at the end clearly indicates that it's a tail extractor.